apps/emqx_schema_validation/src/emqx_schema_validation.erl

```diff
@@ -197,7 +209,75 @@ post_config_update(?VALIDATIONS_CONF_PATH, {delete, Name}, _New, Old, _AppEnvs)
     ok;
 post_config_update(?VALIDATIONS_CONF_PATH, {reorder, _Order}, New, _Old, _AppEnvs) ->
     ok = emqx_schema_validation_registry:reindex_positions(New),
-    ok.
+    ok;
+post_config_update([?CONF_ROOT], {merge, Input}, ResultingConfig, Old, _AppEnvs) ->
+    #{validations := ResultingValidations} = ResultingConfig,
+    #{new_validations := NewValidations0} = prepare_config_merge(Input, Old),
```

Comment:
The `Old` map of `prepare_config_merge/2` has atom as key in pre_config_update, but in post_config_update it is binary. This was not considered in the implementation.
By the way, why is there a need to do merge again in post after it has been done in pre_config_update? 
It seems to be some kind of conflict. Maybe, we could handle nothing in pre_config_update and deal with it directly in post_config_update.

Reply:
> The Old map of prepare_config_merge/2 has atom as key in pre_config_update, but in post_config_update it is binary. This was not considered in the implementation.

Good catch.

> By the way, why is there a need to do merge again in post after it has been done in pre_config_update?
It seems to be some kind of conflict. Maybe, we could handle nothing in pre_config_update and deal with it directly in post_config_update.

Indeed, there's no work needed in pre_config_update.  Thanks!  :beers: